examples/gno.land/r/demo/boards2/permission.gno

```diff
@@ -0,0 +1,35 @@
+package boards2
+
+import (
+	"errors"
+	"std"
+)
+
+const (
+	PermissionBoardCreate  Permission = "board:create"
+	PermissionThreadCreate            = "thread:create"
+	PermissionThreadEdit              = "thread:edit"
+	PermissionThreadDelete            = "thread:delete"
+	PermissionThreadRepost            = "thread:repost"
+	PermissionReplyCreate             = "reply:create"
+	PermissionReplyEdit               = "reply:edit"
+	PermissionReplyDelete             = "reply:delete"
+)
+
+// ErrUnauzorized indicates that user doesn't have a required permission.
+var ErrUnauzorized = errors.New("unauthorized")
+
+type (
+	// Permission defines the type for permissions.
+	Permission string
+
+	// Args is a list of generic arguments.
+	Args []interface{}
+
+	// Permissioner define an interface to for permissioned execution.
+	Permissioner interface {
+		// WithPermission calls a callback when a user has a specific permission.
+		// It panics on error.
+		WithPermission(std.Address, Permission, Args, func(Args))
+	}
```

Comment:
This interface could match `ACL` type's methods but at this point the implementation doesn't use those extra methods. We could add them to this interface once they are used by the boards2 implementation, to keep interface simple for now.

Reply:
Added `HasPermission` to the interface in 1d50806